types/eth_client.go

```diff
@@ -49,3 +61,31 @@ type NoopRPCClient struct{}
 func (c *NoopRPCClient) Call(result any, method string, args ...any) error {
 	return nil
 }
+
+// DialWithRetry attempts to connect to an Ethereum client with retries and exponential backoff.
+// It returns an EthClienter on success or an error if all attempts fail.
+func DialWithRetry(url string, maxRetries int, initialBackoff, maxBackoff time.Duration) (EthClienter, error) {
```

Comment:
Should we add a context here? It seems needed to stop retry if the app fails or the context that is calling this function fails or is canceled. WDYT?